ZipBuilder/Sources/UpdateFirebasePod/main.swift

```diff
@@ -0,0 +1,133 @@
+    }
+  }
+
+  // Override any of the expected versions with the current release manifest, if it exists.
+  if let releasePath = paths.currentReleasePath {
+    let currentRelease = ManifestReader.loadCurrentRelease(fromTextproto: releasePath)
+    print("Overriding the following Pod versions, taken from the current release manifest:")
+    for pod in currentRelease.sdk {
+      releasingVersions[pod.sdkName] = pod.sdkVersion
+      print("\(pod.sdkName): \(pod.sdkVersion)")
+    }
+  }
+
+  if !releasingVersions.isEmpty {
+    print("Updating Firebase Pod in git installation at \(String(describing: paths.gitRootPath!)) " +
+      "with the following versions: \(releasingVersions)")
+  }
+
+  return releasingVersions
+}
+
+private func updateFirebasePod(newVersions: [String: String]) {
+  let podspecFile = paths.gitRootPath! + "/Firebase.podspec"
+  var contents = ""
+  do {
+    contents = try String(contentsOfFile: podspecFile, encoding: .utf8)
+  } catch {
+    print(error)
+    exit(1)
+  }
+  for (pod, version) in newVersions {
+    if pod == "Firebase" {
+      // Replace version in string like s.version = '6.9.0'
+      let range = contents.range(of: "s.version")
+      var versionStartIndex = contents.index(range!.upperBound, offsetBy: 1)
+      while contents[versionStartIndex] != "'" {
+        versionStartIndex = contents.index(versionStartIndex, offsetBy: 1)
+      }
+      var versionEndIndex = contents.index(versionStartIndex, offsetBy: 1)
+      while contents[versionEndIndex] != "'" {
+        versionEndIndex = contents.index(versionEndIndex, offsetBy: 1)
+      }
+      contents.removeSubrange(versionStartIndex...versionEndIndex)
+      contents.insert(contentsOf:"'" + version + "'", at:versionStartIndex)
+    } else {
+      // Replace version in string like ss.dependency 'FirebaseCore', '6.3.0'
+      guard let range = contents.range(of: pod) else {
+        // This pod is not a top-level Firebase pod dependency.
+        continue
+      }
+      var versionStartIndex = contents.index(range.upperBound, offsetBy: 2)
+      while !contents[versionStartIndex].isWholeNumber {
+        versionStartIndex = contents.index(versionStartIndex, offsetBy: 1)
+      }
+      var versionEndIndex = contents.index(versionStartIndex, offsetBy: 1)
+      while contents[versionEndIndex] != "'" {
+        versionEndIndex = contents.index(versionEndIndex, offsetBy: 1)
+      }
+      contents.removeSubrange(versionStartIndex...versionEndIndex)
+      contents.insert(contentsOf:version + "'", at:versionStartIndex)
+    }
+  }
+  do {
+    try contents.write(toFile: podspecFile, atomically: false, encoding: String.Encoding.utf8)
+  }
+  catch {
+    print(error)
+    exit(1)
+  }
+}
+
+do {
+  let newVersions = getExpectedVersions()
+  updateFirebasePod(newVersions: newVersions)
+  print("Updating Firebase pod for version \(String(describing: newVersions["Firebase"]!))")
+
+  // Get the time since the start of the build to get the full time.
```

Comment:
`s/start of the build/start of the tool being run/`

Reply:
Done